test/e2e/network/netpol/network_policy.go

```diff
@@ -123,7 +120,15 @@ var _ = common.SIGDescribe("Netpol", func() {
 
 	ginkgo.Context("NetworkPolicy between server and client", func() {
 		ginkgo.BeforeEach(func() {
-			model = initializeResourcesByFixedNS(f)
+			// These variables control which containers get created in the default model. Create containers with TCP protocol only for tests in this context.
+			var tcp, udp, sctp = true, false, false
+			for _, component := range ginkgo.CurrentGinkgoTestDescription().ComponentTexts {
+				// Create UDP containers along with TCP for this particular test.
+				if component == "should enforce ingress policy allowing any port traffic to a server on a specific protocol [Feature:NetworkPolicy] [Feature:UDP]" {
```

Comment:
The information of whether to use a protocol or not should be provided in the test case itself -- otherwise, the tests will be difficult to comprehend.

Reply:
@mattfenwick ya, that's kind of a chicken and egg problem... we need to pass the information about protocols while creating the model and the logic for that lives in `BeforeEach`. If we move the information about protocol in the test case then we will have to move the model logic there. 

Kind of adding 
```
model = initializeResourcesByFixedNS(f, protocols)
```
to each test case

ref: https://github.com/kubernetes/kubernetes/pull/102354#discussion_r668704334

Reply:
This is a readability vs. convenience tradeoff, and currently this PR is giving away readability for a tiny amount of convenience.

I'm against this approach, for the reason I gave above -- this gives tests surprising behavior which is going to confuse developers.

Since protocols are determined by test case, it makes total sense that each test case has to state which protocols it wants.

You may have been misled by the `addSCTPContainers` flag, which originally was only for developer debugging but then at some point looks like it was wrangled in to determining test behavior.  That never should have happened, so I don't recommend following it as an example.

Reply:
for this test, we don't need to explicitly assert TCP, what about moving the test to the:

- UDP test suite.
- Create a TCP ingress rule.
- Assert UDP should not work.

otherwise we would need to recreted the models again passing TCP/UDP as protocols.

Reply:
Ah yeah we can do that